src/servers/simple.cc

```diff
@@ -63,18 +85,35 @@ ResponseAlloc(
   // releasing the buffer.
 
   // If 'byte_size' is zero just return 'buffer'==nullptr, we don't
-  // need to do any other book-keeping. Only handle allocation in the
-  // CPU region.
-  if ((byte_size == 0) || (memory_type == TRTSERVER_MEMORY_CPU)) {
-    *buffer = (byte_size == 0) ? nullptr : malloc(byte_size);
+  // need to do any other book-keeping.
+  if (byte_size == 0) {
+    *buffer = nullptr;
     *buffer_userp = new std::string(tensor_name);
```

Comment:
*buffer_userp = nullptr when byte_size is 0. Release won't be called (will it?) and buffer_userp is just passed to release.